validator/src/micro.rs

```diff
@@ -216,7 +223,7 @@ impl<TValidatorNetwork: ValidatorNetwork + 'static> NextProduceMicroBlockEvent<T
             if !in_current_state(&head) {
                 None
             } else {
-                let timestamp = head.timestamp() + self.producer_timeout.as_millis() as u64;
+                let timestamp = head.timestamp() + Policy::BLOCK_PRODUCER_TIMEOUT;
```

Comment:
I think this is problematic: Since this always takes the minimum timestamp, it'll factor into the average block production time later. This means that having skip blocks will decrease the average block time, even if the skip blocks get spaced apart further, since the skip block timestamps are incorrect. Perhaps we can use the `self.producer_timeout.as_millis() as u64` here?

Reply:
> I think this is problematic: Since this always takes the minimum timestamp, it'll factor into the average block production time later. This means that having skip blocks will decrease the average block time, even if the skip blocks get spaced apart further, since the skip block timestamps are incorrect.

We talked about this and would fix it in another place: When doing the averaging, don't take skip blocks into account.

> What happens if a skip block takes very long to produce? What happens if the stated timestamp is more off the actual timestamp it got produced than `MAXIMUM_PRODUCER_TIMOUT`?

Producer timeouts only start when a peer receives a block. The actual block timestamp is irrelevant for this, so there's no problem with receiving blocks whose timestamps are far in the past.

Reply:
Fixed